api/jobs/queue.py

```diff
@@ -128,11 +128,10 @@ def retry(job, force=False):
 
         return new_id
 
-
     @staticmethod
-    def enqueue_job(job_map, origin, perm_check_uid=None):
+    def validate_job(job_map, origin, create_job=False, perm_check_uid=None):
```

Comment:
Please do not change this function signature. Instead, revert changes to this function, then remove `job.insert()` to make it consistent with the original documentation. Then check usage of `enqueue_job` to see who will need `job.insert()` added after calling.